libs/ui/library/library.js

```diff
@@ -0,0 +1,166 @@
+    link.setAttribute('href', href);
+    document.head.appendChild(link);
+  }
+}
+
+function getBlockName(block) {
+  const variants = block.className.split('--');
+  let name = variants.shift();
+  name = name.replace(name[0], name[0].toUpperCase());
+  if (variants.length > 0) {
+    variants.push(variants.pop().slice(0, -1));
+    name += ` (${variants.join(', ')})`;
+  }
+  return name;
+}
+
+function getTable(block) {
+  const name = getBlockName(block);
+  const rows = [...block.children];
+  const maxCols = rows.reduce((cols, row) => (
+    row.children.length > cols ? row.children.length : cols), 0);
+  const table = document.createElement('table');
+  const headerRow = document.createElement('tr');
+  headerRow.innerHTML = `<th colspan="${maxCols}">${name}</th>`;
+  table.append(headerRow);
+  rows.forEach((row) => {
+    const tr = document.createElement('tr');
+    [...row.children].forEach((col) => {
+      const td = document.createElement('td');
+      if (row.children.length < maxCols) {
+        td.setAttribute('colspan', maxCols);
+      }
+      td.innerHTML = col.innerHTML;
+      tr.append(td);
+    });
+    table.append(tr);
+  });
+  return { name, table: table.outerHTML };
+}
+
+async function loadBlockList(path, list) {
+  const resp = await fetch(`${repoDomain}${path}`);
+  if (!resp.ok) return;
+  const json = await resp.json();
+  json.data.forEach(async (blockGroup) => {
+    const pageResp = await fetch(`${repoDomain}${blockGroup.value}.plain.html`);
+    if (!pageResp.ok) return;
+    const html = await pageResp.text();
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(html, 'text/html');
+    const blocks = doc.body.querySelectorAll('div[class]');
+    blocks.forEach((block) => {
+      const item = document.createElement('li');
+      const name = document.createElement('p');
+      const converted = getTable(block);
+      name.textContent = converted.name;
+      const { table } = converted;
+      const copy = document.createElement('button');
+      copy.addEventListener('click', (e) => {
+        e.target.classList.add('copied');
+        setTimeout(() => { e.target.classList.remove('copied'); }, 3000);
+        const blob = new Blob([table], { type: 'text/html' });
+        createCopy(blob);
+      });
+
+      item.append(name, copy);
+      list.append(item);
+    });
+  });
+}
+
+function stub(list) {
+  const item = document.createElement('li');
+  const name = document.createElement('p');
+  name.textContent = 'This library isn\'t supported';
+  item.append(name);
+  list.append(item);
+}
+
+function loadList(type, list) {
+  list.innerHTML = '';
+  switch (type.text) {
+    case 'Blocks':
+      loadBlockList(type.path, list);
+      break;
+    case 'Templates':
+      stub(list);
+      break;
+    case 'Placeholders':
+      stub(list);
+      break;
+    case 'Tokens':
+      stub(list);
+      break;
+    default:
+      console.log('no list');
```

Comment:
Is this a good LANA opportunity?

Reply:
I have to ask @chrischrischris some questions on the best way to do this. I'll have another PR to fix it tomorrow.